---
title: Design KASP primers with Primer3 v2.5.0
author: Junli Zhang
date: '2021-12-31'
slug: design-primers-with-primer3
categories:
  - Research
tags:
  - KASP
  - primer3
draft: false
---

This tool uses [primer3](https://github.com/primer3-org/primer3) to design KASP primers for given flanking sequences of SNPs/indels. If the genome of your species has gene duplicates (such as wheat), please try [SNP_Primer_Pipeline2](https://github.com/pinbo/SNP_Primer_Pipeline2) for gene/sub-genome specifc primers.

Input file is a text file with each line looks like below (at least **50bp** on each side of the SNP). An example input can be downloaded [here](/files/KASP-example-input.txt).  
`SNP1 ATAATGTTAGCAGGGGTA[C/G]ACTGGCTGCTTTTGTATTCAAA`  
`SNP2 TGGTTCATGCATATGTTG[CTGT/-]GTGTGCATGCATTGCAGGG`

<h4>Step 1: load the SNP file: each line is a SNP/indel</h4>
<div id="options" style="font-size:90%;color:blue;">
<label for="snpfile">with 2 columns separated with space or tab: unique SNP name (no space allowed), flanking sequences</label><br>
<input id="snpfile" type="file"><br>

<textarea id="paste" name="paste" rows="6" cols="85" placeholder="OR paste your sequences here"></textarea><br>
<p id="demoFq" style="display:none;"></p>

[Primer Size](https://primer3.ut.ee/primer3web_help.htm#PRIMER_MIN_SIZE)
Min <input size="2" id="PRIMER_MIN_SIZE" value="18" type="text">
Opt <input size="2" id="PRIMER_OPT_SIZE" value="20" type="text">
Max <input size="2" id="PRIMER_MAX_SIZE" value="30" type="text">

[Primer Tm](https://primer3.ut.ee/primer3web_help.htm#PRIMER_MIN_TM)
Min <input size="2" id="PRIMER_MIN_TM" value="58.0" type="text">
Opt <input size="2" id="PRIMER_OPT_TM" value="60.0" type="text">
Max <input size="2" id="PRIMER_MAX_TM" value="65.0" type="text">
[Max Tm Difference](https://primer3.ut.ee/primer3web_help.htm#PRIMER_PAIR_MAX_DIFF_TM)
     <input size="2" id="PRIMER_PAIR_MAX_DIFF_TM" value="2.0" type="text">

[Product Size Ranges](https://primer3.ut.ee/primer3web_help.htm#PRIMER_PRODUCT_SIZE_RANGE)
    <input size="40" id="PRIMER_PRODUCT_SIZE_RANGE" value="40-70 70-100 100-120 120-150" type="text">

[Max Hairpin](https://primer3.ut.ee/primer3web_help.htm#PRIMER_MAX_HAIRPIN_TH)
     <input size="2" id="PRIMER_MAX_HAIRPIN_TH" value="47.0" type="text">

[Number of Primers To Return](https://primer3.ut.ee/primer3web_help.htm#PRIMER_NUM_RETURN)
<input size="4" id="PRIMER_NUM_RETURN" value="1" type="text"> for each SNP

<input type="checkbox" id="PRIMER_PICK_ANYWAY" value="1"> [Pick primers anyway](https://primer3.ut.ee/primer3web_help.htm#PRIMER_PICK_ANYWAY) even if it violates specific constraints

<input type="checkbox" id="addTail" name="addTail" value="1">
<label for="addTail">Add FAM/VIC tails to the allele specific primers</label>

<p id="help"></p>
</div>
<h4>Step 2: Start designing primers</h4>
<button onclick="designPrimer()">Start Designing</button>


<div id="download-btn" style="display:none">
    <h4>Step 3: Download designed primers</h4>
    <button id="download" onclick="download()">Download the result (a csv file)</button><br><br>
</div>
<p id="error" style="color:red;"></p>
<pre><code id="stdout"></code></pre>


## Help

This app will design N (number to return) KASP for both the forward and reverse flanking sequences of each SNP, so 2N of KASPs for each SNP.

Each KASP has 3 primers named as "SNP1", "SNP2" and "common". SNP1 and SNP2 only have the 3' end SNP difference. Please add FAM and VIC tails to the 5' end of the two primers before ordering if you did not check the option above to include the tails in the output.

```
FAM    GAAGGTGACCAAGTTCATGCT
VIC    GAAGGTCGGAGTCAACGGATT
```

### Output explanation
The output is the same as the [primer3 website](https://primer3.ut.ee/) output. Here I just explain a few confusing names of the output based on the [primer3 help page](https://primer3.ut.ee/primer3web_help.htm):

- [SELF_ANY](https://primer3.ut.ee/primer3web_help.htm#PRIMER_MAX_SELF_ANY): the tendency of a primer to bind to itself (interfering with target sequence binding).
- [SELF_END](https://primer3.ut.ee/primer3web_help.htm#PRIMER_MAX_SELF_END): the ability to use itself as a target and amplify a short piece (forming a primer-dimer).
- [PAIR_COMPL_ANY](https://primer3.ut.ee/primer3web_help.htm#PRIMER_PAIR_MAX_COMPL_ANY): the tendency of the left primer to bind to the right primer.
- [PAIR_COMPL_END](https://primer3.ut.ee/primer3web_help.htm#PRIMER_PAIR_MAX_COMPL_END): the ability to bind the 3'-END of the left primer to the right primer.

To me the most important parameter is Tm (close to 60 C and small difference between the left primer and the common primer) and hairpin (the smaller the better). Other parameter can help choose better primers when there are more options.

### Caveat for indels
This app seeks the first nucleotide (nt) that is different from the two alleles, and uses that nt as the SNP to design KASP, so the allele specific primers still only have 1 nt difference. Actually, there are more options for indels than SNPs. Depending on the size of the indels, you could design quite different allele-specific primers.

### When no primers with the default settings
- set the Max hairpin up to 70; and/or
- set the Max Tm difference up to 8.0 C; and/or
- set the primer Tm from 56 to 70; and/or
- set the primer length from 17 to 33; or just
- check "pick primers anyway"


## Acknowledgement
1. **primer3**: https://github.com/primer3-org/primer3


<!-- <script src="https://cdn.biowasm.com/v2/aioli/latest/aioli.js"></script> -->
<script src="/tools/aioli/latest/aioli.js"></script>
<script src="/libs/FileSaver.min.js"></script>
<script src="/libs/primer3.js"></script>