<!DOCTYPE html>
<html>
  <head>
    <title>Junli Zhang - CV</title>
    <meta charset="utf-8">
    <link rel="stylesheet" href="/libs/cv.css" type="text/css" />
    <link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.2.0/css/all.css" integrity="sha384-hWVjflwFxL6sNzntih27bfxkr27PmbbK/iSvJ+a4+0owXq79v+lsFkW54bOGbiDQ" crossorigin="anonymous">
    <link rel="stylesheet" href="https://cdn.rawgit.com/jpswalsh/academicons/master/css/academicons.min.css">
  </head>


  <body id="div1">
    <div class="article-content" id="content">
    </div>
    <textarea class="hideDontTakeUpSpace" id="source">
# Zhang, Junli

Assistant Professor, Texas A&M AgriLife Research  
6500 Amarillo Blvd West, Amarillo, TX 79106  
<i class="far fa-envelope"></i> <junli.zhang@ag.tamu.edu> &amp;nbsp;
<i class="fa fa-phone"></i> 806-677-5633 &amp;nbsp;
<i class="fab fa-github"></i> [pinbo](https://github.com/pinbo) &amp;nbsp;
<i class="ai ai-google-scholar"></i> [citation](https://scholar.google.com/citations?user=0ZZ8xGYAAAAJ&hl=en)

# Professional Experience
.entry[
## Sep 2025 - Present
  - Assistant Professor in Small Grain Genetics  
  - *Texas A&M AgriLife Research, Amarillo, Texas, USA*

## Nov 2018 - Aug 2025
  - Assistant Project Scientist  
  - Dr. Jorge Dubcovsky’s lab, Department of Plant Sciences  
  - *University of California, Davis, California, USA*

## Nov 2013 - Oct 2018
  - Postdoctoral Scholar  
  - Dr. Jorge Dubcovsky’s lab, Department of Plant Sciences  
  - *University of California, Davis, California, USA*
]

# Education
.entry[
## Aug 2008 - Dec 2013
  - Ph.D. Plant Science
  - Dr. Jianli Chen’s lab, Dept. of Plant, Soil & Entomological Sciences
  - *University of Idaho, Moscow, Idaho, USA*

## Sep 2004 - Jul 2008
  - B.S. Agronomy
  - College of Agronomy
  - *Northwest A&F University, Shaanxi, China*
]

# Research Experience
*Nov 2013-Aug 2025. Postdoc & Assistant Project Scientist. University of California, Davis*
- Spatial and single-cell expression analyses of early wheat spike development
- Developing A second Generation Regulatory Region Capture Panel for Wheat
- Promoter capture sequencing of Kronos TILLING mutants
- Cloning and Characterization of Plant Height Gene *Rht25*
- Yield QTL Pyramiding to Improve Elite Wheat Cultivars
- Fine Mapping and Cloning Genes Related to Drought Tolerance and Yield Components in Wheat
- Genome-wide Association Study on Water Use Efficiency and Yield Components of Wheat Using a Spring Wheat Association Mapping Panel and Nested Association Mapping Panel
- Genome-wide Association Mapping of Stripe Rust Resistance in Spring Hexaploid Wheat from National Small Grains Collection (NSGC) 

*Aug 2008-Oct 2013. Research Assistant. University of Idaho*

- Association Mapping of Hagberg Falling Number in Hard White Spring Wheat
- Water and Nitrogen Use Efficiency Study of Triticeae CAP (TCAP) Wheat and Barley Collections (Canopy spectrum reflectance and genome-wide association mapping)
- Quantitative Trait Locus (QTL) Mapping of Grain Yield and Physiological Traits Using a Population of Recombinant Inbred Lines of Common Wheat
- Study on Barley Yellow Dwarf Virus (BYDV) Resistance of Bread Wheat by RNAi Technique Using Both Virus-Induced Gene Silencing (VIGS) and Transgenic Methods

# Professional Skills
- Strong background on genetics, genomics and bioinformatics.
- Plant breeding, field experiment design and management, and marker-assisted selection
- Quantitative trait locus (QTL) mapping: linkage mapping and association mapping
- Fine mapping and cloning
- Experiment design and statistical analysis
- Next Generation Sequencing (NGS) and analysis
- Gene Editing with CRISPR/Cas9 (sgRNA design, protoplast testing, CRISPR sequencing etc.)
- Common lab procedures used in molecular genetics
- Protein interaction (Y2H, BiFC, and CoIP)
- DNA marker development (e.g. KASP marker)
- EMS mutation creation and selection
- Strong computer skills (know Linux platform well)
- Computer languages: Perl, Python, R, Shell Script
- Good at Microsoft Office and statistical software R and SAS
- Good at MapMaker 3.0b, Windows Cartographer, TASSEL, GAPIT, R/qtl

# Teaching Experience
*Guest Lecturer, UC-Davis, Jan 2018: Marker development in polyploid wheat*

*Guest Lecturer, UC-Davis, Jan 2018: Introduction of the wheat EcoTilling website*

*Training on Bioinformatic tools, UC-Davis, every one or two years*

# Mentoring Experience

**Zhang Lab**
- Yahya Rauf, Postdoctoral Associate (Sep 2025 - Present)

**Dubcovsky Lab**

- Federico Mirone, visiting Graduate Student from University of Milan (2023)
- Dr. Hongchun Xiong, Visiting scholar from Chinese Academy of Agricultural Sciences (CAAS) (May 2019 - May 2020)
- Priscilla Glenn, Graduate Student (2017-2019)
- Saarah Kuzay, Graduate Student (2016-2018)
- Dmitry Vaisman, visiting Graduate Student from University of Haifa (2018)
- Yana Olifir, undergraduate student & Junior Specialist (2015-2020)
- Brendan McCarthy-Sinclair, undergraduate student (2014-2015)
- Laurel Hoffman, Junior Specialist (2013-2014)

# Tools Developed
- [Web Apps of bioinformatics](https://junli.netlify.app/apps/): a collection of bioinformatics tools
- [Find wheat gene homeologs and their homologs in Arabidopsis, rice and barley](https://junli.netlify.app/apps/wheat-homeolog/)
- [SNP Primer Design Pipeline](https://github.com/pinbo/SNP_Primer_Pipeline): design KASP and CAPS/dCAPS primers for SNP genotyping
- [CRISPR-wheat](https://github.com/pinbo/CRISPR-wheat): design CRISPR guide RNAs in wheat
- [CRISgo](https://github.com/pinbo/CRISgo): process NGS data of CRISPR editings
- [A shiny app for generating barcode labels](https://junli.shinyapps.io/label_barcodes/)

# Professional Affiliation
- American Society of Agronomy (ASA)
- Crop Science Society of America (CSSA)

# Professional Service
**Reviewer**: Genome Biology, New Phytologist, Plant Genome, Agronomy, BMC Plant Biology, Euphytica, Genes & Genomics, Theoretical and Applied Genetics, Agriculture, Genes, Plants, G3: Genes | Genomes | Genetics, Journal of Experimental Botany, Plant Disease, Journal of Cereal Science, Functional Integrative Genomics.

# Honors and Awards
- **2011-2012**    Easton, Gene and Marlene PSES Scholarship
- **2010-2011**    Pure Line Scholarship & Toevs, John L. & Lois K. Scholarship

# Publications

1. Xu, X., H. Lin, **J. Zhang**, G. Burguener, F. Paraiso, et al. 2025. Spatial and single-cell expression analyses reveal complex expression domains in early wheat spike development. bioRxiv: 2025.02.15.638402. (**First 3 authors contributed equally**)

1. Rottersman, M.G., W. Zhang, **J. Zhang**, G. Grigorean, G. Burguener, et al. 2025. Deletion of wheat alpha-gliadins from chromosome 6D improves gluten strength and reduces immunodominant celiac disease epitopes. Theor Appl Genet 138(5): 94.

1. **Zhang, J.**, G.F. Burguener, F. Paraiso, and J. Dubcovsky. 2024. Natural alleles of LEAFY and WAPO1 interact to regulate spikelet number per spike in wheat. Theor Appl Genet 137(11): 257.

1. **Zhang, J.**, H. Xiong, G.F. Burguener, H. Vasquez-Gross, Q. Liu, et al. 2023. Sequencing 4.3 million mutations in wheat promoters to understand and modify gene expression. PNAS 120(38): e2306494120.

2. **Zhang, J.**, C. Li, W. Zhang, X. Zhang, Y. Mo, et al. 2023. Wheat plant height locus *RHT25* encodes a PLATZ transcription factor that interacts with DELLA (RHT1). PNAS 120(19): e2300203120.

3. **Zhang, J.**, J.M. Debernardi, G.F. Burguener, F. Choulet, E. Paux, et al. 2023. A second-generation capture panel for cost-effective sequencing of genome regulatory regions in wheat and relatives. The Plant Genome 16(1): e20296.

4. Gabay, G., H. Wang, **J. Zhang**, J.I. Moriconi, G.F. Burguener, et al. 2023. Dosage differences in 12-OXOPHYTODIENOATE REDUCTASE genes modulate wheat root growth. Nat Commun 14(1): 539.

5. Glenn, P., D.P. Woods, **J. Zhang**, G. Gabay, N. Odle, et al. 2023. Wheat bZIPC1 interacts with FT2 and contributes to the regulation of spikelet number per spike. Theor Appl Genet 136(11): 237.

6. **Zhang, J.** 2022. Check CRISPR Editing Events in Transgenic Wheat with Next-Generation Sequencing. In: Wani, S.H. and Kumar, A., editors, Genomics of Cereal Crops. Springer US, New York, NY. p. 95-106.

7. Dang, C., **J. Zhang**, and J. Dubcovsky. 2022. High-resolution mapping of Yr78, an adult plant resistance gene to wheat stripe rust. The Plant Genome: e20212.

8. Moriconi, J.I., M. Silva, **J. Zhang**, G.E. Tranquilli, and G.E. Santa-María. 2022. A genome-wide association study unveils key chromosome regions involved in determining sodium accumulation in wheat under conditions of low potassium supply. Journal of Plant Physiology 275: 153739.

9. Chen, Y., Y. Liu, **J. Zhang**, A. Torrance, N. Watanabe, et al. 2022. The Triticum ispahanicum elongated glume locus P2 maps to chromosome 6A and is associated with the ectopic expression of SVP-A1. Theor Appl Genet 135: 2313-2331.

10. Kuzay, S., H. Lin, C. Li, S. Chen, D.P. Woods, **J. Zhang**, et al. 2022. WAPO-A1 is the causal gene of the 7AL QTL for spikelet number per spike in wheat. PLOS Genetics 18(1): e1009747.

11. Gabay, G., **J. Zhang**, G.F. Burguener, T. Howell, H. Wang, et al. 2021. Structural rearrangements in wheat (1BS)–rye (1RS) recombinant chromosomes affect gene dosage and root length. The Plant Genome 14(1): e20079. (**First 2 authors contributed equally**)

12. Glenn, P., **J. Zhang**, G. Brown-Guedira, N. DeWitt, J.P. Cook, et al. 2021. Identification and characterization of a natural polymorphism in FT-A2 associated with increased number of grains per spike in wheat. Theor Appl Genet 135: 679-692.

13. Qiao, L., J. Wheeler, R. Wang, K. Isham, N. Klassen, W. Zhao, M. Su, **J. Zhang**, J. Zheng and J. Chen. 2021. Novel Quantitative Trait Loci for Grain Cadmium Content Identified in Hard White Spring Wheat. Frontiers in Plant Science 12: 2672.

14. Kuzay, S., Y. Xu, **J. Zhang**, A. Katz, S. Pearce, Z. Su, M. Fraser, J. A. Anderson, G. Brown-Guedira, N. DeWitt, A. P. Haugrud, J. D. Faris, E. Akhunov, G. Bai, J. Dubcovsky. 2019. Identification of a candidate gene for a QTL for spikelet number per spike on wheat chromosome arm 7AL by high-resolution genetic mapping. Theor Appl Genet 132: 2689-2705. (**First 3 authors contributed equally**)

15. **Zhang, J.**, S. A. Gizaw, E. Bossolini, J. Hegarty, T. Howell, A. Carter, E. Akhunov, J. Dubcovsky. 2018. Identification and validation of QTL for grain yield and plant water status under contrasting water treatments in fall-sown spring wheats. Theor Appl Genet 131: 1741-1759.

16. Dong, Z., J. M. Hegarty, **J. Zhang**, W. Zhang, S. Chao et al. 2017. Validation and characterization of a QTL for adult plant resistance to stripe rust on wheat chromosome arm 6BS (*Yr78*). Theor Appl Genet 130: 2127-2137. (**First 3 authors contributed equally**)

17. Liu, Y., **J. Zhang**, Y.-G. Hu, and J. Chen. 2017. Dwarfing genes *Rht4* and *Rht-B1b* affect plant height and key agronomic traits in common wheat under two water regimes. Field Crops Research 204: 242-248.

18. Wang, R., J. Chen, J. A. Anderson, **J. Zhang**, W. Zhao et al. 2017. Genome-wide association mapping of Fusarium Head Blight resistance in spring wheat lines developed in the Pacific Northwest and CIMMYT. Phytopathology 107: 1486-1495.

19. Chen, J., M.J. Guttieri, **J. Zhang**, D. Hole, E. Souza, and B. Goates. 2016. A novel QTL associated with dwarf bunt resistance in Idaho 444 winter wheat. Theor Appl Genet. 129: 2313-2322.

20. Chen, J., J. Wheeler, K. O’Brien, W. Zhao, N. Klassen, **J. Zhang**, B. Bowman, Y. Wang, C. Jackson, J. M. Marshall and X.M. Chen. 2016. Registration of 'UI Platinum' hard white spring wheat. Journal of Plant Registrations 10: 36-40. 

21. Bulli, P., **J. Zhang**, S. Chao, X. Chen, and M. Pumphrey. 2016. Genetic architecture of resistance to stripe rust in a global winter wheat germplasm collection. G3: Genes, Genomes, Genetics 6: 2237-2253.

22. Bowman, B.C., J. Chen, **J. Zhang**, J. Wheeler, Y. Wang, W. Zhao, S. Nayak, N. Heslot, H. Bockelman, and J.M. Bonman. 2015. Evaluating grain yield in spring wheat with canopy spectral reflectance. Crop Science 55: 1881-1890. 

23. Bonman, J.M., E.M. Babiker, A. Cuesta-Marcos, K. Esvelt-Klos, G. Brown-Guedira, S. Chao, D. See, J. Chen, E. Akhunov, **J. Zhang**, H.E. Bockelman, T.C. Gordon. 2015. Genetic diversity among wheat accessions from the USDA National Small Grains Collection. Crop Science 55: 1243-1253.

24. Maccaferri, M., **J. Zhang**, P. Bulli, Z. Abate, S. Chao, D. Cantu, E. Bossolini, X. Chen, M. Pumphrey, J. Dubcovsky. 2015. A Genome-Wide Association Study of Resistance to Stripe Rust (*Puccinia striiformis* f. sp. *tritici*) in a Worldwide Collection of Hexaploid Spring Wheat (*Triticum aestivum* L.). G3: Genes, Genomes, Genetics 5: 449-465. (**First 3 authors contributed equally**)

25. **Zhang, J.**, J. Chen, C. Chu, W. Zhao, J. Wheeler, E.J. Souza, and R.S. Zemetra. 2014. Genetic dissection of QTL associated with grain yield in diverse environments. Agronomy 4: 556-578.

26. **Zhang, J.**, J. Chen, B.C. Bowman, K. O’Brien, J.M. Marshall, J.M. Bonman. 2014. Association mapping of Hagberg falling number in hard white spring wheat. Crop Science 54: 1243-1252.

27. Chen, J., G. Hu, **J. Zhang**, C. Chu, Y. Wu. 2014. Mapping of STS markers developed from drought tolerance candidate genes and preliminary analysis of their association with yield-related traits in common wheat (*Triticum aestivum*). Cereal Research Communications 42: 199-208. 

28. Chen, J., J. Wheeler, J. Clayton, W. Zhao, K. O’Brien, **J. Zhang**, C. Jackson, J.M. Marshall, B.D. Brown, K. Campbell, X.M. Chen, R. Zemetra, and E.J. Souza. 2013. Registration of 'UI Stone' soft white spring wheat. Journal of Plant Registrations 7: 1-6.

29. Li, P., J. Chen, P. Wu, **J. Zhang**, C. Chu, D. See, G. Brown-Guedira, R. Zemetra, and E. Souza. 2011. Quantitative trait loci analysis for the effect of *Rht-B1* dwarfing gene on coleoptile length and seedling root length and number of bread wheat. Crop Science 51: 2561-2568.


# Invited Talks

1. **Zhang, J.**, H. Xiong, G.F. Burguener, H. Vasquez-Gross, Q. Liu, et al. Sequencing 4.3 million mutations in wheat promoters to understand and modify gene expression. The Plant & Animal Genome Conference (PAG31), San Diego, CA, January 12-17, 2024.

2. **Zhang, J.**. Wheat RHT25 encodes a PLATZ transcription factor that interacts with DELLA (RHT1). The International Forum on Triticeae Genomics and Molecular Breeding (Phase II in 2023), May 19, 2023.

3. **Zhang, J.**. Accessing natural and induced variation in the regulatory regions of the wheat genes. 5TH VIRTUAL DURUM MEETING, May 11, 2023.

4. **Zhang, J.**, C. Li, W. Zhang, X. Zhang, Y. Mo, et al. Cloning and Characterizing Wheat Reduced Plant Height Gene Rht25. The Plant & Animal Genome Conference (PAG30), San Diego, CA, January 13-18, 2023.

5.  **Zhang, J.**, J.M. Debernardi, G.F. Burguener, F. Choulet, E. Paux, et al. 2023. A Second Generation Capture Panel for Cost-Effective Sequencing of Genome Regulatory Regions in Wheat and Relatives. The Plant & Animal Genome Conference (PAG30), San Diego, CA, January 13-18, 2023.


# Posters

1. **Zhang, J.**, G. F. Burguener, J. M. Debernardi, F. Choulet, E. Paux, J. Enk, J. Dubcovsky. 2022. A New Target Enrichment Panel for Comprehensive and Cost-Effective Genomic Regulatory Element Sequencing in Wheat. Poster presentation and abstract in the proceedings of 2022 Plant and Animal Genome XXIX Conference, San Diego, CA, Jan. 08-12.

1. Gabay, G., T. R. Howell, **J. Zhang**, T. Fahima, A. Lukaszewski, J. Dubcovsky. 2020. Characterization of a 1RS chromosome with a 1BS introgression associated with seminal root length and root development in Wheat. Poster presentation and abstract in the proceedings of 2020 Plant and Animal Genome XXVIII Conference, San Diego, CA, Jan. 11-15.

1. **Zhang, J.**, S. Gizaw, E. Bossolini, M. Maccaferri, J. Hegarty, A. H. Carter, S. Chao, E. Akhunov and J. Dubcovsky. 2017. Identification and validation of QTL for yield components under contrasting water treatments in a collection of spring wheat from North America. Poster presentation and abstract in the proceedings of 2017 Plant and Animal Genome XXV Conference, San Diego, CA, Jan. 13-18.

1. Hegarty, J., Z. Dong, **J. Zhang**, W. Zhang, S. Chao, X. Chen and J. Dubcovsky. 2017. Validation and characterization of two QTL for adult plant resistance to stripe rust on wheat chromosome arms 6DS (*Yr77*) and 6BS (*Yr78*). Poster presentation and abstract in the proceedings of 2017, Plant and Animal Genome XXV Conference, San Diego, CA, Jan. 13-18.

1. **Zhang, J.**, S. Abate Gizaw, E. Bossolini, M. Maccaferri, J. M. Hegarty, A. H. Carter, S. Chao, E. Akhunov, J. Dubcovsky. 2016. Identification and validation of QTL for canopy water status under terminal drought in a collection of spring wheat from North America. Poster presentation and abstract in the proceedings of 2016, Plant and Animal Genome XXIV Conference, San Diego, CA, Jan. 08-13.

1. Bulli, P., **J. Zhang**, S. Chao, X.M. Chen, M. Pumphrey. Population Structure, Genetic Diversity, and Stripe rust resistance in core subset of the USDA-ARS winter wheat germplasm collection. 2015. Poster presentation and abstract in the proceedings of 2015 ASA, CSSA and SSSA Int'l Annual meeting, Minneapolis, MN, Nov. 15-18.

1. Chen, J., **J. Zhang**, W. Zhao, J. Wheeler, N. Klassen, and J. Anderson. Assessment of resistance to Fusarium head blight in spring wheat lines grown in the Pacific Northwest and CIMMYT. 2015. Poster presentation and abstract in the proceedings of 2015 ASA, CSSA and SSSA Int'l Annual meeting, Minneapolis, MN, Nov. 15-18.

1. Maccaferri, M., **J. Zhang**, P. Bulli, Z. Abate, S. Chao, D. Cantu, E. Bossolini, X. Chen, M. Pumphrey, J. Dubcovsky. A genome-wide association study of resistance to stripe rust (*Puccinia striiformis* f. sp. *tritici*) in a worldwide collection of hexaploid spring wheat (*Triticum aestivum* L.). 2015. International Plant & Animal Genome XXIII, San Diego, CA, Jan. 10-14. (**Invited Talk and Poster**)

1. **Zhang, J.**, J. Chen, C. Chu, W. Zhao, J. Wheeler, E.J. Souza, and R.S. Zemetra. 2013. Genetic dissection of QTL associated with grain yield in diverse environments. Poster presentation and abstract in the proceedings of 2013 ASA, CSSA and SSSA Int'l Annual meeting, Tampa, FL, Nov. 3-6.

1. **Zhang, J.**, Chen, J., Y. Wang, B. Bonman, J. Wheeler, W, Zhao, K. O’Brien, J.M. Marshall, H. Bockelman, and J. Bonman. 2012. Association mapping of low falling number in hard white spring wheat collection materials. Poster presentation and abstract in the proceedings of 2012 ASA, CSSA and SSSA Int'l Annual meeting, Cincinnati, OH, Oct. 21-24.

1. **Zhang, J.**, J. Chen, W, Zhao, J. Wheeler, E. Souza, and R. Zemetra. 2012. Quantitative trait loci associated with canopy temperature, chlorophyll content index, and flag leaf senescence in a recombinant inbred line population of winter wheat (*Triticum aestivum* L.). Poster presentation and abstract in the proceedings of 2012 International Plant and Animal Genome meeting, San Diego, California, Jan. 14-18.

1. **Zhang, J.**, J. Chen, Ch. Chu, P. Li, W. Zhao, J. Clayton, J. Wheeler, E. Souza, and R. Zemetra. 2011. QTL analysis in a RIL population indicated effects of increasing wheat grain yield through coordinated improvements on yield-related traits. Poster presentation and abstract in the proceedings of 2011 NAPB (National Association of Plant Breeders), annual conference, College Station, TX, May 23-25.

1. **Zhang, J.**, J. Chen, P. Li, W. Zhao, E. Souza, and R. Zemetra. 2010. Quantitative trait loci for grain yield in a recombination inbred line population of wheat (*Triticum aestivum* L.). Poster presentation and abstract in the proceedings of Plant Breeding for Drought Tolerance, Fort Collins, CO, June 24-25.

1. **Zhang, J.**, P. Li, W. Zhao, Ch. Chu, E. Souza, and Robert Zemetra, and J. Chen. 2010. Identifying quantitative trait loci for canopy temperature, chlorophyll content, and grain yield in a recombination inbred line population of wheat (*Triticum aestivum* L.). Poster presentation and abstract in the proceedings of 2010 International Plant and Animal Genome meeting, San Diego, California, Jan. 10-13.

    </textarea>
    <script src="/libs/marked.min.js"></script>
  <script src="/libs/remark-converter.js"></script>

	<script>
		var slide = convert(document.getElementById("source").value);
		var content = document.getElementById("content");
		content.innerHTML += slide;
		/*remove textarea */
		var parent = document.getElementById("div1");
		var child = document.getElementById("source");
		parent.removeChild(child);
	</script>
  </body>
</html>
