---
title: Fst from AMOVA and pairwise Fst
author: Junli Zhang
date: '2017-11-27'
slug: fst-from-amova-and-pairwise-fst
categories:
  - R
tags:
  - Tips
---

## AMOVA
Today I figured out that we could get the overall Wrightâ€™s Fixation index (*Fst*) from Analysis of molecular variance (AMOVA). Here is an example to do AMOVA and calculate the *Fst*.

```{r, warning=FALSE,message=FALSE}
library(pegas)
require(ape)
data(woodmouse)
d <- dist.dna(woodmouse)
g <- factor(c(rep("A", 7), rep("B", 8)))
p <- factor(c(rep(1, 3), rep(2, 4), rep(3, 4), rep(4, 4)))
amova(d ~ g/p, nperm = 100) # 2 levels
m1 <- amova(d ~ g/p, nperm = 100)
```

Based on the explanation on [Wikipedia](https://en.wikipedia.org/wiki/Fixation_index), *Wright's definition illustrates that Fst measures the amount of genetic variance that can be explained by population structure.*

```{r}
# overall Fst
Fst <- with(m1$varcomp, sum(sigma2[1:(length(sigma2)-1)])/sum(sigma2))
Fst
```

Here the two population structure g and p do not have significant variance, so almost close to 0. Still not sure why it can calculate some negative variance.

## Pairwise Fst
Fst traditionally is calculated in Arlequin, but its input file is so difficult to prepare, so I still prefer doing it in R. After comparison, I found R package "StAMPP" has very close results with Arlequin.

```{r}
# import genotype data and convert to allele frequecies
library(StAMPP)
library(lattice)

## Write a panel function (after examining 'args(panel.levelplot) to add text in the plot
myPanel <- function(x, y, z, ...) {
  panel.levelplot(x,y,z,...)
  panel.text(x, y, ifelse(x<y, round(z,2), ""), col = "red")
}

data(potato.mini, package="StAMPP")
potato.freq <- stamppConvert(potato.mini, "r")
# Calculate pairwise Fst values between each population
potato.fst <- stamppFst(potato.freq, 100, 95, 1)

## color borrowed from Arlequin
ColorRamp <- colorRampPalette(c("white", "steelblue1", "blue3"))
fst = potato.fst$Fsts
#fst = fst[nrow(fst):1,] # invert rows
tt = levelplot(t(fst),panel = myPanel, col.regions=ColorRamp,xlab="Population",ylab="Population",main="Pairwise Fst")
print(tt)
```

I work on allohexaploid wheat. Although it is a hexaploid, but it performs like a diploid, so coding of the SNP data is like AA, BB, AB.